src/components/GradeForm.js

```diff
@@ -0,0 +1,82 @@
+import React, { Component } from 'react';
+import Grade from './Grade';
+
+class GradeForm extends Component {
+
+    state = {
+        label: 'JavaScript Quiz',
+        score: '3',
+        grades: [
+            { label: 'Important Test', score: 20},
+            { label: 'Nuclear Energy Exam', score: 1}
+        ]
+
+    }
+    handleChangeLabel = e => {
+        this.setState({
+            label: e.target.value
+        });
+    }
+    handleChangeScore = e => {
+        this.setState({
+            score: e.target.value,
+
+        });
+    }
+    handleSubmit = (e) => {
+        e.preventDefault();
+        let pass
+
+        if (this.state.score >= 12) {
+            pass = true
+        }
+    
+        const newItem = {
+          label: this.state.label,
+          score: this.state.score,
+          passing: pass
+          
+        };
+    
+        // return an object that gives new value (to set state)
+        this.setState(prev => {
+          const newList = prev.grades.slice(0);
+          newList.push(newItem);
+          return {
+            grades: newList
+          };
+    
+        });
+      }
+
+
+    render() {
+        let pass
+
+        if (this.state.score >= 12) {
+            pass = true
+        }
+
+        return (
+            <div className="GradeForm">
+                <Grade label={this.state.label} score={this.state.score} total={20} passing={pass}></Grade>
+                <p>
+                    <form onSubmit={this.handleSubmit}>
+                        <label>Label</label>
+                        <input value={this.state.label} onChange={this.handleChangeLabel} ></input>
+                        <label>Score</label>
+                        <input value={this.state.score} onChange={this.handleChangeScore} type="range" id="points" name="points" min="0" max="20"></input>
+                    <p><button type="submit">Add</button></p>
+                    </form>
+                    
+                </p>
+                
+                {this.state.grades.map(data => {
+                    const passInLoop = data.score >= 12;
+                    return <Grade label={data.label} score={data.score} total={20} passing={passInLoop}/>
```

Comment:
Whenever you `map` a series of components into a list like this, React needs a `key` associated with each element so it can track it. Without a key, React will be forced to re-render this List View each time `grades` changes (this list is small, so it's not a big deal, but imagine if the list was 1000s entries long!). It's a good idea to get into the habit.

A key should be a concatenated string that uniquely identifies this row _and_ it's state, so something like ``${data.label}_${data.score}_${passInLoop}`` might be a good option.